### PR TITLE
#1833 - event sponsorship

### LINKED ISSUE #1833: Changes to event sponsorship
This is part of Membership project.

1. Switch `event.sponsors` to Foreign Key pointing at particular organisation
2. Remove Sponsorship model
3. Change calculations of completed workshops in Membership from host to sponsor

### PR DESCRIPTION
**Pull request by pbanaszkiewicz**
Fixes #1833.

### CODE REVIEW

File: amy/workshops/views.py [hunk 1]
```diff
@@ -105,7 +104,7 @@
     Person,
     Role,
     Membership,
-    Sponsorship,
+    # ,
```
Comment (lauryndbrown) [inline amy/workshops/views.py:107]: Nit, do we still need this comment?

File: docs/models_in_applications.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ The new application structure contains:
 `EventSubmission`, `DCSelfOrganizedEventRequest`, ~~`ProfileUpdateRequest`~~,
 ~~`InvoiceRequest`~~ - by the way all these models are deprecated and to be
 removed in future - `WorkshopRequest`, and `TrainingRequest`),
-* `fiscal` - for organizations, memberships, and sponsorships,
+* `fiscal` - for organizations, and memberships,
```
Comment (lauryndbrown) [inline docs/models_in_applications.md:25]: Nit I think you can removed the comma there:
```
`fiscal` - for organizations and memberships,
```

File: amy/workshops/forms.py [hunk 3]
```diff
@@ -1064,11 +1057,17 @@ class EventsMergeForm(forms.Form):
     start = forms.ChoiceField(choices=TWO, initial=DEFAULT, widget=forms.RadioSelect)
     end = forms.ChoiceField(choices=TWO, initial=DEFAULT, widget=forms.RadioSelect)
     host = forms.ChoiceField(choices=TWO, initial=DEFAULT, widget=forms.RadioSelect)
+    sponsor = forms.ChoiceField(choices=TWO, initial=DEFAULT, widget=forms.RadioSelect)
     administrator = forms.ChoiceField(
         choices=TWO,
         initial=DEFAULT,
         widget=forms.RadioSelect,
     )
+    public_status = forms.ChoiceField(
```
Comment (lauryndbrown) [inline amy/workshops/forms.py:1066]: I don't quite understand what caused this change. Was public status not originally a part of the form? Is that new functionality for events?

**Review by lauryndbrown — APPROVED** [review]
Looks good to me. I just had a few nits. It's a fairly large change so definitely feel free to point out anything in particular you wanted feedback on if I missed it.

File: amy/workshops/forms.py (hunk 3 above)
Comment (pbanaszkiewicz) [reply amy/workshops/forms.py:1066]: Yes. It's a miss from a few weeks back when this field was introduced. Thankfully this form is not being used too often. Good catch.

**Pull request merged**